lisa/util/shell.py

```diff
@@ -38,18 +38,22 @@ def wait_tcp_port_ready(
     timout_timer = create_timer()
     while timout_timer.elapsed(False) < timeout:
         with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as tcp_socket:
-            result = tcp_socket.connect_ex((address, port))
-            if result == 0:
-                is_ready = True
-                break
-            else:
-                if times % 10 == 0 and log:
-                    log.debug(
-                        f"cannot connect to TCP port: {port}, error code: {result}, "
-                        f"tried times: {times}, elapsed: {timout_timer}. retrying..."
-                    )
-                sleep(1)
-                times += 1
+            try:
+                result = tcp_socket.connect_ex((address, port))
+                if result == 0:
+                    is_ready = True
+                    break
+                else:
+                    if times % 10 == 0 and log:
+                        log.debug(
+                            f"cannot connect to {address}:{port}, "
+                            f"error code: {result}, tried times: {times},"
+                            f" elapsed: {timout_timer}. retrying..."
+                        )
+                    sleep(1)
+                    times += 1
+            except Exception as e:
```

Comment:
Well, we can add that, but we cannot add address:port to each line of log. This line can be corelated by environment name like `customized_0`. It's a global unique environment name, search it can get all information.

Reply:
This is a fatal failure, most of the time, so it won't be repetitive. How is this related to env. name, I fail to graps that. BTW, what's the reasoning behind that "customized" naming?

Reply:
There may be better name than "customized", feel free to suggest name to differentiate environments between test case requirements and explicit defined in runbook.
For the log line, it's ok to print IP/port here. Some day, there may be other information is necessary, like user name, kernel version. We cannot print all of them on one line. So it needs to look for the log by `customized_0`. The key word corelated logs.

Reply:
So customized_0 is for the env name? How about "unnamed_env_0", then? It's best to have some namespacing on the thing, so readers are not so lost at a first glance.

Reply:
The key information is about where an environment is from. Both runbook defined and generated from test cases may be unnamed. So unnamed may not be enough. In most place, the name shows like `env[customized_0]`, so the type is out of it. Other types like platform, node, notifier... They won't be a part of name.